services/common/snmp/rsu41/rsu_message_forward.py

```diff
@@ -0,0 +1,325 @@
+import subprocess
+import logging
+import common.snmp.snmpcredential as snmpcredential
+import common.snmp.snmperrorcheck as snmperrorcheck
+import common.util as util
+import common.snmp.rsu_message_forward_helpers as rsu_message_forward_helpers
+from datetime import datetime
+
+# SNMP property to string name and processing function
+# Supports SNMP RSU 4.1 Spec
+prop_namevalue = {
+    "iso.0.15628.4.1.7.1.2": (
+        "Message Type",
+        rsu_message_forward_helpers.message_type_rsu41,
+    ),
+    "iso.0.15628.4.1.7.1.3": ("IP", rsu_message_forward_helpers.ip_rsu41),
+    "iso.0.15628.4.1.7.1.4": ("Port", int),
+    "iso.0.15628.4.1.7.1.5": ("Protocol", rsu_message_forward_helpers.protocol),
+    "iso.0.15628.4.1.7.1.6": ("RSSI", int),
+    "iso.0.15628.4.1.7.1.7": ("Frequency", int),
+    "iso.0.15628.4.1.7.1.8": (
+        "Start DateTime",
+        rsu_message_forward_helpers.startend_rsu41,
+    ),
+    "iso.0.15628.4.1.7.1.9": (
+        "End DateTime",
+        rsu_message_forward_helpers.startend_rsu41,
+    ),
+    "iso.0.15628.4.1.7.1.10": ("Forwarding", rsu_message_forward_helpers.fwdon),
+    "iso.0.15628.4.1.7.1.11": ("Config Active", rsu_message_forward_helpers.active),
+}
+
+
+def ip_to_hex(ip):
+    hex_dest_ip = ""
+    for octet in ip.split("."):
+        if len(hex(int(octet))[2:]) == 1:
+            hex_dest_ip += "0"
+        hex_dest_ip += hex(int(octet))[2:]
+    return "00000000000000000000FFFF" + hex_dest_ip
+
+
+# delta is in years
+def hex_datetime(now, delta=0):
```

Comment:
question: looks like this functionality is in a couple places, should we refactor?

Reply:
Addressed. I broke this into the rsu_message_forward_helpers.py script since it really is only used for message forwarding anyways.